games-action/postal/patches/postal-1.0.0.patchset

```diff
@@ -0,0 +1,446 @@
+From 73945da945fa31bf535df1708460359a0518d3f5 Mon Sep 17 00:00:00 2001
+From: Craig Watson <watsoncraigjohn@gmail.com>
+Date: Thu, 2 Sep 2021 11:10:02 +0100
+Subject: Haiku support
+
+
+diff --git a/RSPiX/Src/ORANGE/File/file.cpp b/RSPiX/Src/ORANGE/File/file.cpp
+index ca6cfef..610ab1a 100644
+--- a/RSPiX/Src/ORANGE/File/file.cpp
++++ b/RSPiX/Src/ORANGE/File/file.cpp
+@@ -149,6 +149,11 @@
+ #include <ctype.h>
+ #endif
+ 
++#if __HAIKU__
++#include <FindDirectory.h>
++#include <Path.h>
++#endif
++
+ #ifdef WIN32
+ #define WIN32_LEAN_AND_MEAN 1
+ #include <windows.h>
+@@ -405,11 +410,21 @@ extern const char *FindCorrectFile(const char *_pszName, const char *pszMode)
+             if (prefpath[strlen(prefpath)-1] != '/') strcat(prefpath, "/");
+ 
+             strcat(prefpath, "Library/Application Support/Postal Plus/");
+-            #else
++            #else            
+             const char *homedir = getenv("HOME");
+             const char *xdghomedir = getenv("XDG_DATA_HOME");
+             const char *append = "";
+ 
++#ifdef __HAIKU__
++			BPath dataPath;
++			//haiku paths are not set if launched outside of console
++			if(find_directory(B_USER_NONPACKAGED_DATA_DIRECTORY, &dataPath) == B_OK) 
++			{
++				xdghomedir = dataPath.Path();
++			}
++			homedir = NULL;
++#endif
++
+             if (xdghomedir == NULL)
+             {
+                 if (homedir == NULL)
+diff --git a/RSPiX/Src/ORANGE/GameLib/Shapes.h b/RSPiX/Src/ORANGE/GameLib/Shapes.h
+index b71c2e9..5d92fe0 100644
+--- a/RSPiX/Src/ORANGE/GameLib/Shapes.h
++++ b/RSPiX/Src/ORANGE/GameLib/Shapes.h
+@@ -61,6 +61,10 @@
+ #ifndef SHAPES_H
+ #define SHAPES_H
+ 
++#ifdef __HAIKU__
++#include <stdint.h>
++#endif
++
+ 
+ class RPt
+ {
+diff --git a/main.cpp b/main.cpp
+index 813d910..ad7248e 100644
+--- a/main.cpp
++++ b/main.cpp
+@@ -154,6 +154,12 @@
+ #include "WishPiX/Menu/menu.h"
+ #endif
+ 
++#ifdef __HAIKU__
++#include <Entry.h>
++#include <FindDirectory.h>
++#include <Path.h>
++#endif
++
+ ////////////////////////////////////////////////////////////////////////////////
+ // Macros/types/etc.
+ ////////////////////////////////////////////////////////////////////////////////
+@@ -765,6 +771,44 @@ int main(int argc, char **argv)
+         return 1;
+     #endif
+ 
++#ifdef __HAIKU__
++		//pref file should be copied from data if missing
++		//will then be used from user non-packaged for all writes
++		BPath systemDataPath, userDataPath;
++		if(find_directory(B_USER_NONPACKAGED_DATA_DIRECTORY, &userDataPath) == B_OK) 
```

Comment:
the directory won't be created by default. 

https://git.haiku-os.org/haiku/tree/headers/os/storage/FindDirectory.h#n206

Reply:
The USER_NONPACKAGED_DATA_DIRECTORY?

The PostalPlus folder is created on launch if the app is run with nothing else, but the user will need to copy the files across as instructed for any of it to work anyway.

Reply:
it's irrelevant because the code under doesn't even check whether the folder PostalPlus exists.

Reply:
BTW have you a reason not to use B_USER_SETTINGS_DIRECTORY ?

Reply:
It's irrelevant? What are you talking about?  The game won't work at all without the data files being copied over.  This is the same for loads of ports.  If the folder is missing and therefore the data files are missing then the game doesn't work.  Would you like me to also code something in so it can access your GOG or Steam account to download all the game assets and install them for you?

There is code elsewhere that creates the PostalPlus folder, etc.  All I'm doing is working around the packaging system and conventions which you insist on.  I doubt you have looked at the code at all as a whole and are simply commenting on patch, recipe, which is useless.  There are checks elsewhere in the code.  All I have done is mimic the existing functionality and rework some stuff so it'll run at all on Haiku.

This type of feedback is useless.  One line comments don't inform anything, linking to a header for a function?  Is this some ego thing?

It does not use the B_USER_SETTINGS_DIRECTORY because it's not a Haiku app, but a port of a game from the mid-90s where such things didn't exist.  There's a few functions which literally all the file system calls in the program use which would take too much time and effort to rework for my level of interest.

If you want to rework it so it makes separate calls for prefs file, save games, etc. and can then follow the Haiku conventions you are welcome to.  But I would start with the 90% of Haiku Ports that also don't follow the conventions properly.

I have better things to do than tit for tat with you.

Reply:
Please calm down. People reviewing your code need to make sure they understand what you are doing, and so they ask questions. It is not an attack on what you did.

We will collectively be responsible for maintaining this, and so we need to understand how it works. So we make sure of that before merging it. That's all.

Reply:
If you want people to contribute then you need to make meaningful feedback.  Telling me something is "irrelevant" seems needlessly aggressive and unhelpful but I need to calm down.

I don't understand why I seem to be singled out by certain individuals for additional scrutiny.  This is not the first time and makes me wonder why I bother at all.

Comments as above are not meaningful, useful, etc. and show that the commenter hasn't done much other than a shallow review.  If you want people to work with this project then put some effort in.  If there are additional rules, conventions, etc. that need to be followed then these should be documented up front.  Frankly the majority of my base for recipes is from finding other recipes that look similar and adapting them.  Then I am told that something in a similar recipe is not correct.  This is the basis for my work since the documentation is not really good enough.

If it is wrong then you can simply point at the rule that is documented.  Meanwhile I can point at all the other ports where this was apparently fine and wonder why I merit special attention.

Reply:
Instead of specifying a constant path to resource files, I would make a launcher (see Arx-Libertatis, Julius, dhewm3 for example). It will be very easy to add support for a new game to this launcher - https://github.com/EXL/BeGameLauncher

![screenshot48](https://user-images.githubusercontent.com/2559260/132300841-8543b899-22cb-433c-8c02-27988b6c881c.png)

Reply:
The main problem with this is the game almost hardcodes everything in to using the PostalPlus folder.  It has one function it calls for everything that effectively assumes everything will sit under that folder.  You can redirect some things within the INI file but will later hit issues with it loading from the file system direct and it won't get past the load screen when starting a new level.  Effectively it requires some sort of constant path because it's basically hardcoded to one - the only thing I've done is make it use a Haiku friendly path.

To do things differently would require reworking that function and also detecting the type of file it is trying to access.

There's also no point in storing the POSTAL.INI somewhere else since it will recreate it under the PostalPlus folder anyway.

These are all issues that can be resolved by someone else if they have the interest.

Reply:
I'm asking because a standard user won't look for a settings file in USER_NONPACKAGED_DATA_DIRECTORY.

I take this is acceptable to everyone then.